tap_appsflyer/__init__.py

```diff
@@ -147,11 +147,19 @@ def parse_source_from_url(url):
                       giveup=giveup,
                       factor=2)
 @utils.ratelimit(10, 1)
-def request(url, params=None):
+def request(url, api_token, params=None):
 
     params = params or {}
     headers = {}
 
+    if len(api_token) > 36:
+        if re.match(re.compile('(^[\w-]+\.[\w-]+\.[\w-]+$)'), api_token):
+            headers["Authorization"] = "Bearer " + api_token
+        else:
+            raise Exception("Invalid AppsFlyer V2 token")
+    else:
+        params["api_token"] = api_token
```

Comment:
```suggestion
    if re.match(re.compile('(^[\w-]+\.[\w-]+\.[\w-]+$)'), api_token):
         headers["Authorization"] = "Bearer " + api_token
    else:
         params["api_token"] = api_token      
```
Is there any doc which says v1 token would never go beyond 36 characters. If not, I would have my code like this. If the token is invalid then API would throw appropriate error.

Reply:
Since v1 is not a JWT token and V2 is a JWT token, you can start the if condition with checking for JWT validation and get rid of length of string condition(which we are not sure of)